Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -647,546 +552,351 @@ protected virtual void Dispose(bool disposing)
+        /// <param name="bindPreference">The published server URL that matches the source address.</param>
         /// <returns><c>true</c> if a match is found, <c>false</c> otherwise.</returns>
-        private bool MatchesPublishedServerUrl(IPObject source, bool isInExternalSubnet, out string bindPreference, out int? port)
+        private bool MatchesPublishedServerUrl(IPAddress source, bool isInExternalSubnet, out string bindPreference)
         {
             bindPreference = string.Empty;
-            port = null;
+            int? port = null;
+
+            var validPublishedServerUrls = _publishedServerUrls.Where(x => x.Key.Address.Equals(IPAddress.Any)
+                                                                            || x.Key.Address.Equals(IPAddress.IPv6Any)
+                                                                            || x.Key.Subnet.Contains(source))
+                                                                .GroupBy(x => x.Key)
+                                                                .Select(x => x.First())
+                                                                .OrderBy(x => x.Key.Address.Equals(IPAddress.Any)
+                                                                            || x.Key.Address.Equals(IPAddress.IPv6Any))
+                                                                .ToList();
 
             // Check for user override.
-            foreach (var addr in _publishedServerUrls)
+            foreach (var data in validPublishedServerUrls)
             {
-                // Remaining. Match anything.
-                if (addr.Key.Address.Equals(IPAddress.Broadcast))
-                {
-                    bindPreference = addr.Value;
-                    break;
-                }
-                else if ((addr.Key.Address.Equals(IPAddress.Any) || addr.Key.Address.Equals(IPAddress.IPv6Any)) && isInExternalSubnet)
+                // Get address interface.
+                var intf = _interfaces.OrderBy(x => x.Index).FirstOrDefault(x => data.Key.Subnet.Contains(x.Address));
+
+                if (isInExternalSubnet && (data.Key.Address.Equals(IPAddress.Any) || data.Key.Address.Equals(IPAddress.IPv6Any)))
                 {
                     // External.
-                    bindPreference = addr.Value;
+                    bindPreference = data.Value;
                     break;
                 }
-                else if (addr.Key.Contains(source))
+
+                if (intf?.Address != null)
                 {
-                    // Match ip address.
-                    bindPreference = addr.Value;
+                    // Match IP address.
+                    bindPreference = data.Value;
                     break;
                 }
             }
 
             if (string.IsNullOrEmpty(bindPreference))
             {
+                _logger.LogDebug("{Source}: No matching bind address override found.", source);
```

Comment:
Log messages should never end with a dot
```suggestion
                _logger.LogDebug("{Source}: No matching bind address override found", source);
```